### ISSUE TITLE
UTC Dates in Mutation Input fields are broken in 0.14.7 and upards

### ISSUE THREAD

**Issue by lngr** [graphql-dotnet/graphql-dotnet#286]
The recent changes to the DateTime output (allow DateTime in StringType-fields, https://github.com/graphql-dotnet/graphql-dotnet/commit/67bf2ed015753801787f9f13931be7ba0c3b7186 ) unfortunately breaks UTC dates again for input fields/query arguments.

More specifically, we have the following resolver for a Relay mutation:

```
.Resolve(args =>
                {
                    var input = args.GetArgument<MyInputType>("input");
                    ...
```

looking at the source of `ResolveFieldContext.GetArgument`, it will call `arg.GetPropertyValue<TType>();` in 
https://github.com/graphql-dotnet/graphql-dotnet/blob/master/src/GraphQL/Types/ResolveFieldContext.cs#L85

Here it will eventually call `ConvertValue()` in https://github.com/graphql-dotnet/graphql-dotnet/blob/master/src/GraphQL/ObjectExtensions.cs#L105
which was unfortunately modified in the pull request.

Since the test against `DateTime` has been removed in https://github.com/graphql-dotnet/graphql-dotnet/pull/272/files#diff-ffca66d73ea1b01c7a2ab35b3b63a4f9L126 , the string is converted via "ToString() again, which was to root case in #197, see the fix in
https://github.com/graphql-dotnet/graphql-dotnet/pull/198

Since `DateTime.ToString()` removes the `Z`-suffix, the "UTC" information is lost again.

TBH, if someone wants to send date strings in `StringGraphType`, I'd recommend to do the conversion manually before feeding the object to GraphQL and not let the framework handle such use cases. I therefore recommend to revert the changes in https://github.com/graphql-dotnet/graphql-dotnet/pull/272

**Comment by joemcbride** [comment]
Sorry about this!  Dates suck.  Looks like we missed a Unit test for that scenario.  You may be right that it will be difficult to support both of these use cases.  I would opt for just using DateTime as well.  I say lets get a Unit Test written for the `GetArgument` scenario and see where that leads.

**Comment by lngr** [comment]
No harm caused 😄  I tried to come up with a unit test that catches the bug but was not successful yet.

**Comment by cstlaurent** [comment]
Hi, @lngr, I worked with @mamasselin on the pull request #272 ans I would advise against reversing it. The goal of this merge request was to fix the unexpected behavior of receiving `null` in your `StringGraphType` field value if you had a string that happened to look like an ISO date. I think this correction puts this library in line with the GraphQL standard.

Date handling is not in the standard since it's not native to json, but I still agree with you that losing timezone information when you have a `DateGraphType` is very bad.

I was pretty sure we covered the issue you mentioned while doing the modifications but maybe we missed something.

Anyway, @mamasselin and me are looking into it.
